src/main/features/requirements/controller/ca-teamScale.ts

```diff
@@ -65,15 +101,19 @@ export const CA_POST_TEAM_SCALE = async (req: express.Request, res: express.Resp
   const { projectId } = req.session;
   const assessmentId = req.session.currentEvent.assessmentId;
   const dimension = req.session.dimensions;
-  const scalabledata = req.session.scaledata;
+  //const scalabledata = req.session.scaledata;
   //let data=parseInt(scalabledata);
   const scalabilityData = dimension.filter(data => data.name === 'Scalability')[0];
- if(req.body.team_option!=undefined){
+  const assessmentDetail = await GET_ASSESSMENT_DETAIL(SESSION_ID, assessmentId);
+  const Dimensionrequirements=assessmentDetail.dimensionRequirements;
+  const dimensionName=scalabilityData['name'];
+  const weight=Dimensionrequirements.filter(x=>x["dimension-id"]===4)[0].weighting;
```

Comment:
1. const weight=Dimensionrequirements?.filter(x=>x["dimension-id"]===4)?.[0]?.weighting;
2. use above line

Reply:
these are implemented by dhivya in her latest commit